airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io.airbyte.cdk/read/CdcPartitionsCreatorFactory.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.read
+
+import io.airbyte.cdk.command.OpaqueStateValue
+import io.micronaut.core.annotation.Order
+import jakarta.inject.Singleton
+
+@Singleton
+@Order(10)
+class CdcPartitionsCreatorFactory(
+    val cdcContext: CdcContext,
+    val concurrencyResource: ConcurrencyResource,
+) : PartitionsCreatorFactory {
+
+    override fun make(stateQuerier: StateQuerier, feed: Feed): PartitionsCreator? {
+        val opaqueStateValue: OpaqueStateValue? = stateQuerier.current(feed)
+        return when (feed) {
+            is Global -> {
+                // TODO: remove the following if-block
+                if (opaqueStateValue != null) {
+                    // For the time being, only do one round
+                    return null
+                }
```

Comment:
I believe this replicates the existing and temporary behaviour of doing one round of CDC reading and then that's it for the `Global` feed.